ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -249,8 +253,33 @@ public void setEventListener(ReactRootViewEventListener eventListener) {
     return Assertions.assertNotNull(mJSModuleName);
   }
 
-  /* package */ @Nullable Bundle getLaunchOptions() {
-    return mLaunchOptions;
+  public @Nullable Bundle getAppProperties() {
+    return mAppProperties;
+  }
+
+  public void setAppProperties(@Nullable Bundle appProperties) {
```

Comment:
To avoid inconsistencies, let's assert that this method is UIThread only as well? (`UiThreadUtil.assertOnUiThread()`)